### PR TITLE
 Fix bug: Use try-with-resources or close this "Statement" in a "finally" clause.

### PR DESCRIPTION
**Pull request by Jave-Chen**
## What is the purpose of the pull request
#1701 Resources should be closed

## Brief change log
1. Use try-with-resources with Statement and ResultSetk.
2. Fix some code smell.

### CODE REVIEW

**Comment by Jave-Chen** [comment]
Quality Gate failed

Failed condition B Maintainability Rating on New Code (is worse than A)
Failed condition 0.0% 0.0% Coverage on New Code (is less than 80%)
Failed condition 47.4% 47.4% Duplicated Lines on New Code (is greater than 3%)

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java [hunk 1]
```diff
@@ -243,46 +233,35 @@ private void runScript(Connection conn, Reader reader , String dbName) throws IO
...
+                                    String name = md.getColumnLabel(i);
+                                    logger.info("{} \t", name);
+                                }
+                                logger.info("");
+                                while (rs.next()) {
+                                    for (int i = 0; i < cols; i++) {
+                                        String value = rs.getString(i);
+                                        logger.info("{} \t", value);
+                                    }
+                                    logger.info("");
```
Comment (Technoboy-) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java]: remove this line

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java [hunk 2]
```diff
@@ -243,46 +233,35 @@ private void runScript(Connection conn, Reader reader , String dbName) throws IO
...
+                                }
+                            }
+                        }   
+                    } catch (SQLException e) {
+                        logger.error(e.getMessage(),e);
```
Comment (Technoboy-) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java]: logger.error("SQLException", e);

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java [hunk 3]
```diff
@@ -291,11 +270,11 @@ private void runScript(Connection conn, Reader reader , String dbName) throws IO
 			}
 
 		} catch (SQLException e) {
-			logger.error("Error executing: " + sql);
+			logger.error("Error executing: {}", sql);
 			throw e;
 		} catch (IOException e) {
 			e.fillInStackTrace();
```
Comment (Technoboy-) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java]: remove

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java [hunk 4]
```diff
@@ -207,11 +197,11 @@ private void runScript(Connection conn, Reader reader) throws IOException, SQLEx
 			}
 
 		} catch (SQLException e) {
-			logger.error("Error executing: " + command.toString());
+			logger.error("Error executing: {}", command);
 			throw e;
 		} catch (IOException e) {
 			e.fillInStackTrace();
```
Comment (Technoboy-) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java]: remove

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java [hunk 5]
```diff
@@ -161,44 +161,34 @@ private void runScript(Connection conn, Reader reader) throws IOException, SQLEx
...
+                                int cols = md.getColumnCount();
+                                for (int i = 0; i < cols; i++) {
+                                    String name = md.getColumnLabel(i);
+                                    logger.info("{} \t", name);
+                                }
+                                logger.info("");
+                                while (rs.next()) {
+                                    for (int i = 0; i < cols; i++) {
+                                        String value = rs.getString(i);
+                                        logger.info("{} \t", value);
```
Comment (Technoboy-) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java]: add more info

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java [hunk 6]
```diff
@@ -161,44 +161,34 @@ private void runScript(Connection conn, Reader reader) throws IOException, SQLEx
...
+                                    }
+                                    logger.info("");
+                                }
+                            }
+                        }   
+                    } catch (SQLException e) {
+                        logger.error(e.getMessage(),e);
```
Comment (Technoboy-) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ScriptRunner.java]: logger.error("SQLException", e);

**Comment by Jave-Chen** [comment]
Roger

**Pull request merged**